[excel] (IA) Break out "basic" samples into independent articles (#644)
OfficeDev/office-scripts-docs: docs/resources/samples/table-samples.md

---
title: Table samples

> description: A collection of samples how to interact with Excel tables.

Reviewer: ```suggestion
description: A collection of samples showing how to interact with Excel tables.
```

  const selectedSheet = workbook.getActiveWorksheet();

  // Create a table with the used cells.
  const usedRange = selectedSheet.getUsedRange();
  const newTable = selectedSheet.addTable(usedRange, true);

  // Sort the table using the first column.
  newTable.getSort().apply([{ key: 0, ascending: true }]);
}
```

## Filter a table

This sample filters an existing table using the values in one of the columns.

```TypeScript
function main(workbook: ExcelScript.Workbook) {
  // Get the table in the workbook named "StationTable".
  const table = workbook.getTable("StationTable");

  // Get the "Station" table column for the filter.
  const stationColumn = table.getColumnByName("Station");

  // Apply a filter to the table that will only show rows 
  // with a value of "Station-1" in the "Station" column.
  stationColumn.getFilter().applyValuesFilter(["Station-1"]);
}
```

> [!TIP]
> Copy the filtered information across the workbook by using `Range.copyFrom`. Add the following line to the end of the script to create a new worksheet with the filtered data.
>
> ```TypeScript
>   workbook.addWorksheet().getRange("A1").copyFrom(table.getRange());
> ```

## Remove table column filters

> This sample removes the filters from a table column, based on the active cell location. The script detects if the cell is part of a table, determines the table column, and clears any filter that are applied on it.

Reviewer: ```suggestion
This sample removes the filters from a table column, based on the active cell location. The script detects if the cell is part of a table, determines the table column, and clears any filters that are applied on it.
```

Download [table-with-filter.xlsx](table-with-filter.xlsx) for a ready-to-use workbook. Add the following script to try the sample yourself!

```TypeScript
function main(workbook: ExcelScript.Workbook) {
  // Get the active cell.
  const cell = workbook.getActiveCell();

  // Get the tables associated with that cell.
  // Since tables can't overlap, this will be one table at most.
  const currentTable = cell.getTables()[0];

>   // If there is no table on the selection, end the script.

Reviewer: ```suggestion
  // If there's no table on the selection, end the script.
```
